github/projects.go

```diff
@@ -466,3 +466,127 @@ func (s *ProjectsService) MoveProjectCard(ctx context.Context, cardID int64, opt
 
 	return s.client.Do(ctx, req, nil)
 }
+
+type ProjectCollaboratorOptions struct {
+	// Permission specifies the permission to grant to the collaborator.
+	// Possible values are:
+	//     read - can read, but not write to or administer this project.
+	//     write - can read and write, but not administer this project.
+	//     admin - can read, write and administer this project.
+	//
+	// Default value "write"
+	// Note that, if you choose not to pass any parameters,
+	// you'll need to set Content-Length to zero when calling out to this endpoint.
+	Permission string `json:"permission,omitempty"`
```

Comment:
Since this field has a default value, let's make this parameter type `*string` instead of `string`.
That way, if the GitHub API adds more options, then this options struct can still be sent, but the `Permission` field does not have to be set.

Reply:
I copied that struct from `repo_collaborators.go` and it used `string` when I use `*string` it leads to a few test failures like 
```
--- FAIL: TestProjectsService_ListCollaborators (0.00s)
        github_test.go:109: Request parameters: map[page:[2] Affiliation:[]], want map[page:[2]]
```

I am not sure why this change would lead to that, it expects the affiliation field to start with a capital letter `Affilliation` and if it isn't set it still expects to have it in the map with an empty value. This wasn't happening before

Reply:
When changing `string` to `*string`, the tests will need to make a string pointer using `String("value")` instead of just `"value"`. See the other existing tests for more examples.

Reply:
Oh I did use ` String("value") ` in the tests, the failure still occurs

Reply:
I apologize... you are correct to leave it as `url:...` since it is a parameter being passed.
Sorry about that.